drivers/include/ph_oem.h

```diff
@@ -0,0 +1,356 @@
+    PH_OEM_IRQ_BOTH     = 0x08, /**< Invert state on new reading (automatically reset) */
+} ph_oem_irq_option_t;
+
+/**
+ * @brief   Calibration option values
+ */
+typedef enum {
+    PH_OEM_CALIBRATE_LOW_POINT  = 0x02,     /**< Low point calibration option */
+    PH_OEM_CALIBRATE_MID_POINT  = 0x03,     /**< Mid point calibration option */
+    PH_OEM_CALIBRATE_HIGH_POINT = 0x04,     /**< High point calibration option */
+} ph_oem_calibration_option_t;
+
+/**
+ * @brief   pH OEM sensor params
+ */
+typedef struct ph_oem_params {
+    i2c_t i2c;                      /**< I2C device the sensor is connected to */
+    uint8_t addr;                   /**< the slave address of the sensor on the I2C bus */
+    gpio_t interrupt_pin;           /**< interrupt pin (@ref GPIO_UNDEF if not defined) */
+    ph_oem_irq_option_t irq_option; /**< behavior of the interrupt pin, disabled by default*/
+} ph_oem_params_t;
+
+/**
+ * @brief   pH OEM interrupt pin callback
+ */
+typedef void (*ph_oem_interrupt_pin_cb_t)(void *);
+
+/**
+ * @brief   pH OEM device descriptor
+ */
+typedef struct ph_oem {
+    ph_oem_params_t params;         /**< device driver configuration */
+    ph_oem_interrupt_pin_cb_t cb;   /**< interrupt pin callback */
+    void *arg;                      /**< interrupt pin callback param */
+} ph_oem_t;
+
+/**
+ * @brief   Initialize a pH OEM sensor
+ *
+ * @param[in,out]   dev      device descriptor
+ * @param[in]       params   device configuration
+ *
+ * @return @ref PH_OEM_OK on success
+ * @return @ref PH_OEM_NODEV if no device is found on the bus
+ * @return @ref PH_OEM_NOT_PH if the device found at the address is not a pH OEM device
+ * @return
+ */
+int ph_oem_init(ph_oem_t *dev, const ph_oem_params_t *params);
+
+/**
+ * @brief   Sets a new address to the pH OEM device by unlocking the
+ *          @ref PH_OEM_REG_UNLOCK register and  writing a new address to
+ *          the @ref PH_OEM_REG_ADDRESS register.
+ *          The device address will also be updated in the device descriptor so
+ *          it is still usable.
+ *
+ *          Settings are retained in the sensor if the power is cut.
+ *
+ *          The address in the device descriptor will reverse to the default
+ *          address you provided through PH_OEM_PARAM_ADDR after the
+ *          microcontroller restarts
+ *
+ * @param[in] dev   device descriptor
+ * @param[in] addr  new address for the device. Range: 0x01 - 0x7f
+ *
+ * @return @ref PH_OEM_OK on success
+ * @return @ref PH_OEM_WRITE_ERR if writing to the device failed
+ */
+int ph_oem_set_i2c_address(ph_oem_t *dev, uint8_t addr);
+
+/**
+ * @brief   Enable the pH OEM interrupt pin if @ref ph_oem_params_t.interrupt_pin
+ *          is defined.
+ *          @note Provide the PH_OEM_PARAM_INTERRUPT_OPTION flag in your
+ *          makefile. Valid options see: @ref ph_oem_irq_option_t.
+ *          The default is @ref PH_OEM_IRQ_BOTH
+ *
```

Comment:
Maybe add a little comment saying that `ph_oem_reset_interrupt_pin` must be called from `cb`.